# 5/01/2020 
### To check the person is eligible for vote or not

In [1]:
age=int(input("enter the age of person "))
if(age>=18 and age<=85):
  print("person is eligible to vote")
else:
  print("person is not eligible to vote")

enter the age of person 50
person is eligible to vote


### To check the sum of two numbers ia 10 or either one of  the  number is 10

In [14]:
x=int(input("enter x value "))
y=int(input("enter y value "))
if(x==10 or y==10 or x+y==10):
  print("True")
else:
  print("false")


enter x value 8
enter y value 2
True


### To print the table with ranges

In [15]:
n=int(input("enter the number "))
l=int(input("enter the lower limit "))
u=int(input("enter the upper limit "))
i=0
while(l<=u):
  i=n*l
  print(n,"*",l,"=",i)
  l=l+1

enter the number 4
enter the lower limit 5
enter the upper limit 8
4 * 5 = 20
4 * 6 = 24
4 * 7 = 28
4 * 8 = 32


### To print the multiplication table

In [12]:
n=int(input("enter the number "))
i=1
while(i<=10):
  print(n,"*",i,"=",n*i)
  i=i+1

enter the number 9
9 * 1 = 9
9 * 2 = 18
9 * 3 = 27
9 * 4 = 36
9 * 5 = 45
9 * 6 = 54
9 * 7 = 63
9 * 8 = 72
9 * 9 = 81
9 * 10 = 90


### To print the sum of natural numbers

In [13]:
n=int(input("enter the number "))
i=1
sum=0
while(i<=n):
  sum=sum+i
  i=i+1
print("sum of natural numbers is ",sum)

enter the number 5
sum of natural numbers is  15


### To print the reverse of a number

In [16]:
n=int(input("enter the number "))
rev=0
while(n!=0):
  rev=rev*10+(n%10)
  n=n//10
print("reverse of a number ",rev)

enter the number 8965
reverse of a number  5698


### To print the number of prime or not

In [18]:
# To check the number is prime or not
n=int(input("enter the number "))
if n > 1:
   for i in range(2,n):
       if (n % i) == 0:
           print(n,"is not a prime number")
           break
   else:
       print(n,"is a prime number")
else:
   print(n,"is not a prime number")

enter the number 9
9 is not a prime number


### To print the number of primes between certain range

In [1]:
l = int(input("enter lower limit "))
u = int(input("enter upper limit"))
print("Prime numbers are:")
for num in range(l, u + 1):
   if num > 1:
       for i in range(2, num):
           if (num % i) == 0:
               break
       else:
           print(num)

enter lower limit 10
enter upper limit50
Prime numbers are:
11
13
17
19
23
29
31
37
41
43
47


### To check the number is palindrome or not

In [19]:
n=int(input("enter the number "))
rev=0
temp=nn
while(n!=0):
  rev=rev*10+(n%10)
  n=n//10
if(temp==rev):
  print("palindrome")
else:
  print("not palindrome")

enter the number 151
palindrome


### To check the number is positive ,negative or zero

In [20]:
n=int(input("enter the number "))
if(n>0):
  print(n," is positive number")
elif(n<0):
  print(n," is negative number")
else:
  print(n," zero")


enter the number 5
5  is positive number


### Factorial of a number

In [21]:
n=int(input("enter the number "))
fact=1
i=1
while(i<=n):
    fact=fact*i
    i=i+1
print("factorial of (",n,") is",fact)
    

enter the number 8
factorial of ( 8 ) is 40320


### Largest of two numbers

In [22]:
a =int(input("enter a value "))
b=int(input("enter b value "))
if a>b:
    print("a is largest")
else:
    print("b is largest")

enter a value 50
enter b value 90
b is largest


### Largest of three numbers

In [23]:
a =int(input("enter a value "))
b=int(input("enter b value "))
c=int(input("enter c value "))
if a>b and a>c:
    print("a is largest")
elif b>c:
    print("b is largest")
else:
    print("c is largest")


enter a value 50
enter b value 70
enter c value 10
b is largest


### Area of circle

In [24]:
r=int(input("enter the radius "))
area=3.14*r*r
print("Area of circle is ",area)

enter the radius 5
Area of circle is  78.5


### Area of triangle

In [25]:
b=int(input("enter the base "))
h=int(input("enter the height "))
area=0.5*b*h
print("Area of triangle is ",area)

enter the base 5
enter the height 9
Area of triangle is  22.5


### squareroot of a number

In [26]:
import math
n=int(input("enter the number "))
print("square root is ",math.sqrt(n))


enter the number 81
square root is  9.0


### Factors of a number

In [27]:
n=int(input("enter the number "))
i=1
while(i<n):
    if(n%i==0):
        print(i)
    i=i+1

enter the number 9
1
3


### Even or Odd number

In [28]:
n=int(input("enter the number "))
if(n%2==0):
    print("even")
else:
    print("odd")

enter the number 56
even


### Divisibility

In [29]:
n=int(input("enter the number "))
if(n%2==0 and n%3==0):
    if(n%4!=0):
        print("number divisible by 2 & 3 but not with 4")
    else:
        print("number divisible by 2  & 3 & 4")
else:
    print("number is divisible by 2 & 3 & 4")

enter the number 24
number divisible by 2  & 3 & 4


### Print numbers using loop

In [30]:
n=int(input("enter the number "))
i=1
while(i<=n):
    print(i)
    i=i+1


enter the number 6
1
2
3
4
5
6


### Print even numbers using loop

In [31]:
n=int(input("enter the number "))
i=1
while(i<=n):
    if(i%2==0):
      print(i)
    i=i+1

enter the number 14
2
4
6
8
10
12
14


### To check the number is strong or not

In [32]:
sum1=0
num=int(input("Enter a number: "))
temp=num
while(num):
    i=1
    f=1
    r=num%10
    while(i<=r):
        f=f*i
        i=i+1
    sum1=sum1+f
    num=num//10
if(sum1==temp):
    print("The number is a strong number")
else:
    print("The number is not a strong number")

Enter a number: 145
The number is a strong number


### To check the number is perfect or not

In [28]:
n = int(input("Enter any number: "))
sum1 = 0
for i in range(1, n):
    if(n % i == 0):
        sum1 = sum1 + i
if (sum1 == n):
    print("The number is a Perfect number!")
else:
    print("The number is not a Perfect number!")

Enter any number: 28
The number is a Perfect number!


### To check the number is armstrong or not

In [2]:
num = int(input("Enter a number: "))
sum = 0
temp = num
while temp > 0:
   digit = temp % 10
   sum += digit ** 3
   temp //= 10
if num == sum:
   print(num,"is an Armstrong number")
else:
   print(num,"is not an Armstrong number")


Enter a number: 135
135 is not an Armstrong number


### To check the number of palindromes in certain range

In [30]:
max = int(input("Enter the Maximum Value : "))
print("Palindrome Numbers are")
for num in range(1, max+ 1):
    temp = num
    reverse = 0
    while(temp > 0):
        Reminder = temp % 10
        reverse = (reverse * 10) + Reminder
        temp = temp //10
    if(num == reverse):
        print("%d " %num, end = '  ')

Enter the Maximum Value : 100
Palindrome Numbers are
1   2   3   4   5   6   7   8   9   11   22   33   44   55   66   77   88   99   

### To print maximum element dynamically

In [2]:
list=[]
n=int(input("enter the number of elements "))
for i in range(0,n):
    k=int(input("enter the number "))
    list.append(k)
ma=list[0]
for i in range(0,n):
    if(ma<list[i]):
        ma=list[i]
print("maximum element of list is ",ma)

enter the number of elements 4
enter the number 50
enter the number 70
enter the number 90
enter the number 80
maximum element of list is  90


### To print the frequency of given numbers

In [5]:
n=int(input("enter the number "))
count=0
while(n>0):
    count=count+1
    n=n//10
print(count)

enter the number 2525
4


# 6/01/2020
### Write a Python program to create a list by concatenating a given list which range goes from 1 to n

In [10]:
my=['p', 'q']
n=int(input())
new=['{}{}'.format(x, y) for y in range(1, n+1) for x in my]
print(new)

5
['p1', 'q1', 'p2', 'q2', 'p3', 'q3', 'p4', 'q4', 'p5', 'q5']


In [3]:
n=int(input())
l=['p','q']
l1=[]
for i in range(1,n+1):
    for j in range(0,len(l)):
        l1.append(l[j]+str(i))
print(l1)

5
['p1', 'q1', 'p2', 'q2', 'p3', 'q3', 'p4', 'q4', 'p5', 'q5']


### Count of  Non Prime Factors

In [2]:
def prime1(n):
    count=0
    for i in range(1,n):
        if(n%i==0):
            count=count+1
    if(count==1):
        return 1
    else:
        return 0
def factors(n):
    count=0
    li=[]
    for i in range(2,n+1):
        if(n%i==0 and i%10):
            li.append(i)
    print(li)
    for i in li:
        x=prime1(i)
        if(x==0):
            count=count+1
    print(count)
n=int(input("enter a number "))
factors(n)

enter a number 100
[2, 4, 5, 25]
2


### Frequency of numbers in list

In [5]:
list=[]
n=int(input("Enter number of elements:"))
for i in range(0,n):
    ele=int(input("Enter element:"))
    list.append(ele)
frequency = {}
for item in list:
   if (item in frequency):
      frequency[item] += 1
   else:
      frequency[item] = 1
for key, value in frequency.items():
   print("% s -> % d" % (key, value))

Enter number of elements:5
Enter element:1
Enter element:2
Enter element:3
Enter element:3
Enter element:1
1 ->  2
2 ->  1
3 ->  2


# 8/01/2020

### To check no of armstrong number between ranges

In [2]:
lower = int(input("Enter lower limit"))
upper = int(input("Enter upper limit ")) 
for num in range(lower,upper + 1):
   sum = 0
   temp = num
   while temp > 0:
       digit = temp % 10
       sum += digit ** 3
       temp //= 10
   if num == sum:
       print(num)

Enter lower limit1
Enter upper limit 1000
1
153
370
371
407


# 11/01/2020

#### Write a Python program to concatenate following dictionaries to create a new one. 
- Sample Dictionary :
  dic1={1:10, 2:20}
  dic2={3:30, 4:40}
  dic3={5:50,6:60}
  Expected Result : {1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}

In [4]:
dic1={1:10, 2:20}
dic2={3:30, 4:40}
dic3={5:50,6:60}
dic4 = {}
for d in (dic1, dic2, dic3): 
    dic4.update(d)
print(dic4)

{1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}


### Write a Python program to check if a given key already exists in a dictionary.
 - d = {1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}

In [10]:
d = {1: 10, 2: 20, 3: 30, 4: 40, 5: 50, 6: 60}
def key(x):
    if x in d:
        print("Key already exist")
    else:
        print("key is not present")
key(5)

Key already exist


Input: a = ["abc", "def", "ghi"]

Output: [['a', 'b', 'c'], ['d', 'e', 'f'], ['g', 'h', 'i']]

In [15]:
a=["abc","def","ghi"]
lis=[]
for i in a:
    k=i.split()
    for j in k:
        u=list(j)
    lis.append(u)
print(lis)

[['a', 'b', 'c'], ['d', 'e', 'f'], ['g', 'h', 'i']]


In [45]:
l=[1,2,3,4,5,6,7,8,9]
x=0
y=9
for i in range(x,y,3):
    x=i
    k=l[x:x+3]
    y=tuple(k)
    print(y)

(1, 2, 3)
(4, 5, 6)
(7, 8, 9)


- The program must accept an integer N as the input. Need to print the output in the following    manner.
     Example Input/Output 1:145  
     Output:
       Five Four One
     Example Input/Output 2:123  
      Output:
       Three two one

In [34]:
n=int(input())
reverse = 0.
rev = 0  
while(n > 0): 
    a = n % 10
    rev = rev * 10 + a 
    n = n // 10       
n1=str(rev)
print(n1)
for i in range(0,len(n1)):
    if(n1[i]=="1"):
        print("one")
    elif(n1[i]=="2"):
        print("two")
    elif(n1[i]=="3"):
        print("Three")
    elif(n1[i]=="4"):
        print("Four")
    elif(n1[i]=="5"):
        print("Five")
    elif(n1[i]=="6"):
        print("Six")
    elif(n1[i]=="7"):
        print("Seven")
    elif(n1[i]=="8"):
        print("Eight")
    elif(n1[i]=="9"):
        print("Nine")
    elif(n1[i]=="0"):
        print("Zero")

143
341
Three
Four
one


- Write a Python program to find the highest 3 values in a dictionary

In [51]:
d={1:10,2:20,3:30,4:40,5:50,6:60}
y=d.values()
k=sorted(y)
print(k[3:6])

[40, 50, 60]


- The program must accept an integer N as the input. Need to print the output in the following manner.
  Boundary Condition(s):
    Example:
      Input1: 1889
      Output:16
      Input2: 19872
      Output:10


In [54]:
n=1889
l= [int(x) for x in str(n)] 
s=0
for i in l:
    if(i%2==0):
        s=s+i
print(s)

16


- The program must accept an integer N as the input. Need to print the output in the following manner.
  Example Input/Output 1:
    Input1:145
    Output:yes
    Input2:123
    Output:no
- Explanation : 
   Individual digit factorial sum is the same as the original number then need to print   YES otherwise  need to print NO
 #145 - 1! + 4! + 5! - 1 + 24 + 120 -- 145 -- YES
 #123 - 1! + 2! + 3! - 1 + 2 + 6 - 9 -- NO

In [55]:
sum1=0
num=int(input("Enter a number:"))
temp=num
while(num):
    i=1
    f=1
    r=num%10
    while(i<=r):
        f=f*i
        i=i+1
    sum1=sum1+f
    num=num//10
if(sum1==temp):
    print("YES")
else:
    print("NO")

Enter a number:145
YES
